dataclasses_json/cfg.py

```diff
@@ -1,11 +1,20 @@
 import functools
-from typing import (Callable, Dict, Optional, Union)
+from enum import auto, Enum
+from typing import (Any, Callable, Dict, Optional, Union)
 
 from marshmallow.fields import Field as MarshmallowField
 
 from dataclasses_json.undefined import Undefined, UndefinedParameterError
 
 
+class EncodeAction(Enum):
```

Comment:
rather than taking an `enum` let's simplify the API and just have it always expect a predicate